### DISCUSSION TITLE
Primary Key ID Increasing value on error requests

### DISCUSSION THREAD

**Discussion by PeterGKafkakis** [supabase/supabase#6557] [Questions]
Currently have a table that set up with 4 columns.

ID(int8), email, phone, zipcode.

Every time I make a request, the id will get autoincremented. So for example, let's say I insert this data below.

I will get id: 1, email: johndoe@gmail.com, phone: 6664442222, zipcode: 22255

Duplicates cant be added into the table, so if i run this 4 times, for example, it still wont insert to the database(unique constraints) BUT if i change the data and insert it, now the id will be at id:5 after 4 error calls. Trying to stop it from autoincrementing the id even on an error call

If i make more requests, even with errors, with the same data, 

const { data, error } = await supabase.from("subscribers").insert([
      {
        email: "johndoe@gmail.com",
        phone: "6664442222",
        zipcode: "22255",
      },
    ]);

    console.log(data);
    console.log(error);

![Error](https://user-images.githubusercontent.com/22582269/165212152-d5566a8d-56be-402b-8b18-2d778ef2858e.png)

**Comment by GaryAustin1** [comment]
The answer is probably similar to this:
https://github.com/supabase/supabase/discussions/6361
